ompi/mca/pml/ucx/pml_ucx.c

```diff
@@ -514,8 +514,9 @@ int mca_pml_ucx_enable(bool enable)
 
 int mca_pml_ucx_progress(void)
 {
-    ucp_worker_progress(ompi_pml_ucx.ucp_worker);
-    return OMPI_SUCCESS;
+    unsigned ret;
+    ret = ucp_worker_progress(ompi_pml_ucx.ucp_worker);
+    return (ret > 0) ? 1 : 0;
```

Comment:
maybe just return whatever  ucp_worker_progress() returns?

Reply:
That was my first approach, too. The documentation for `ucp_worker_progress` states as return value 

> Non-zero if any communication was progressed, zero otherwise. 

Since it returns an `unsigned` the value could be `(unsigned)-1`, which cast to `int` would be seen as signalling an error by the progress loop.

Reply:
@devreal sorry if the documentation of ucp_worker_progress was not clear, but it can't really return -1. It's either 0 or a small integer value which indicated the number of processed events.

Reply:
In that case we can return directly, yes :)